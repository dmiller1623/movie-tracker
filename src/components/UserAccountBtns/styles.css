.userBtns {
  display: flex;
  align-items: center; }
  .userBtns h3 {
    color: #f2f2f2;
    margin: 0 1rem;
    font-size: 2rem; }
  .userBtns button {
    color: #f2f2f2;
    text-decoration: none;
    border: none;
    background-color: Transparent;
    font-size: 1rem;
    margin: 0 2rem 0 0; }
    .userBtns button:hover {
      color: #93A2AA;
      background-color: Transparent; }

.favoritesBtn {
  color: #f2f2f2;
  text-decoration: none;
  margin: 0 .5rem; }
  .favoritesBtn:hover {
    color: #93A2AA; }

span {
  color: #f2f2f2; }

.selected {
  color: #93A2AA; }
