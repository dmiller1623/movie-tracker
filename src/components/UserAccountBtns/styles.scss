.userBtns {
  display: flex;
  align-items: center;
  h3 {
    color: #f2f2f2;
    margin: 0 1rem;
    font-size: 2rem;
  }
  button {
    color: #f2f2f2;
    text-decoration: none;
    border: none;
    background-color: Transparent;
    font-size: 1rem;
    margin: 0 2rem 0 0 ;
    &:hover {
      color: #93A2AA;
      background-color: Transparent;
    }
  }
}

.favoritesBtn {
  color: #f2f2f2;
  text-decoration: none;
  margin: 0 .5rem;
  &:hover {
    color: #93A2AA;
  }
}



span {
  color: #f2f2f2;
}

.selected {
  color: #93A2AA;
}