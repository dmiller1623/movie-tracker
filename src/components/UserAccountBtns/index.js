import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import './styles.css';

const UserAccountBtns = ({ name, handleSignOut }) => {
  return (
    <div className='userBtns'>
      <h3>Hi, {name}!</h3>
      <NavLink activeClassName='selected' className='favoritesBtn' to='/favorites'>
        View Favorites
      </NavLink>
      <span> | </span>
      <button onClick={() => handleSignOut()}>
        Sign Out
      </button>
    </div>
  )
}

export default UserAccountBtns;

UserAccountBtns.propTypes = {
  name: PropTypes.string,
  handleSignOut: PropTypes.func
};
