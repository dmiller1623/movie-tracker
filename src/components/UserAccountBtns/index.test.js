import React from 'react';
import { shallow, mount } from 'enzyme';
import UserAccountBtns from '../UserAccountBtns';
import { BrowserRouter }  from 'react-router-dom';

describe('InitialAccountBtns', () => {

  it('should match the snapshot', () => {
    const name = 'test';
    const wrapper = shallow(
      <BrowserRouter>
        <UserAccountBtns name={name}/>
      </BrowserRouter>);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should call a function handleSignOut() when the sign out button is clicked', () => {
    const mockHandleSignOut = jest.fn();
    const wrapper = mount(
      <BrowserRouter>
        <UserAccountBtns handleSignOut={mockHandleSignOut} />
      </BrowserRouter>);

    wrapper.find('button').simulate('click');
    expect(mockHandleSignOut).toHaveBeenCalled();
  });

});