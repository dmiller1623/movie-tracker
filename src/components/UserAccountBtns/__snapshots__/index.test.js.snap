// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialAccountBtns should match the snapshot 1`] = `"<div class=\\"userBtns\\"><h3>Hi, test!</h3><a class=\\"favoritesBtn\\" href=\\"/favorites\\">View Favorites</a><span> | </span><button>Sign Out</button></div>"`;
